File: new/q.py
class Qubit:

	"""
	Qubit!
	"""

	alpha: complex
	beta: complex

	def __init__(self, value):
		if value:
			alpha = 0j
			beta = 1+0j
		else:
			alpha = 1+0j
			beta = 0j

	def __getitem__(self, index: int):
		return (alpha, beta)[index]

class Register:

	"""
	Quantum register class!
	"""

	def __init__(self, size: int, iv=0) -> None:
		self.qubits = [Qubit(iv) for _ in range(size)]

	def ket(index: int):
		result = 1
		for 

File: new/q.cpp
#include <cmath>
#include <complex>

using namespace std;

int inv(int val) {
	if (val == 1) {
		return 0;
	} else if (val == 0) {
		return 1;
	} else {
		return -1;
	}
}

class Qubit {

	public:

		complex<double> alpha;
		complex<double> beta;

		Qubit(): alpha(complex<double>(1, 0)), beta(complex<double>(0, 0)) {}
		Qubit(int val): alpha(complex<double>(inv(val), 0)), beta(complex<double>(val, 0)) {}

		complex<double> getByIndex(int index) {
			if (index == 0) {
				return alpha;
			} else if (index == 1) {
				return beta;
			} else {
				return complex<double>(0, 0);
			}
		}

		complex<double> operator[](int index) {
			return getByIndex(index);
		}

		int Pr(int index) {
			return pow(getByIndex(index), 2);
		}

};

class QuantumRegister {

	public:

		Qubit* qubits;



};